generator/src/test/java/viper/generator/ConfigurationKeyProcessorTest.java

```diff
@@ -0,0 +1,90 @@
+package viper.generator;
+
+import static viper.generator.ConfigurationKeyProcessor.calculateAnnotationName;
+import static viper.generator.ConfigurationKeyProcessor.calculateConfigurationBeanName;
+
+import org.assertj.core.api.JUnitSoftAssertions;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class ConfigurationKeyProcessorTest {
+
+	private static final String ENUM_CLASS = "EnumBase";
+	@Rule
+	public JUnitSoftAssertions softly = new JUnitSoftAssertions();
+
+	@Test
+	public void shouldFailForNullString_annotationName() throws Exception {
+		softly.assertThatThrownBy(() -> calculateAnnotationName(null, null))
+				.isInstanceOf(IllegalArgumentException.class);
+		softly.assertThatThrownBy(() -> calculateAnnotationName("Clazz", null))
+				.isInstanceOf(IllegalArgumentException.class);
+		softly.assertThatThrownBy(() -> calculateAnnotationName(null, "ClazzAnnot"))
+				.isInstanceOf(IllegalArgumentException.class);
+	}
+
+	@Test
+	public void shouldFailForInvalidString_annotationName() throws Exception {
+		softly.assertThatThrownBy(() -> calculateAnnotationName(ENUM_CLASS, ""))
```

Comment:
What happen if configurationBeanName is not empty and not contains any star symbol?

Reply:
Same as above ;)